SofascoreAcademy/SofascoreAcademy/View/EventViewCell.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  EventViewCell.swift
+//  SofascoreAcademy
+//
+//  Created by Anton Pomper on 10.05.2024..
+//
+
+import UIKit
+
+class EventCell: UICollectionViewCell {
+    
+    var eventView: EventView!
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setupView() {
+        eventView = EventView()
+        contentView.addSubview(eventView)
+        contentView.backgroundColor = .surfaceSurface1
+
+        eventView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+            
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(handleTap))
+        eventView.addGestureRecognizer(tapGesture)
+        eventView.isUserInteractionEnabled = true
+    }
+        
+    @objc func handleTap() {
+        let eventDetailsVC = EventDetailsViewController()
+            if let navigationController = self.findViewController()?.navigationController {
```

Comment:
generalno se ovako nešto ne radi. Ćelija na neki način dojavi parentu da se tap dogodio preko deleagata, action closure ili slično pa onda parent, koji je ujedno i VC, zna odraditi akciju.